AppDB/backup/cassandra_backup.py

```diff
@@ -0,0 +1,308 @@
+      directory)
+    logging.warn("Removing data from {0}".format(data_dir))
+    try:
+      subprocess.Popen('find /opt/appscale/cassandra -name "*" | '
+        'grep ".db\|.txt\|.log" | grep -v snapshot | xargs rm', shell=True)
+      logging.info("Done removing data!")
+    except CalledProcessError as error:
+      logging.error("Error while removing old data from db. Overwriting... "
+        "Error: {0}".format(str(error)))
+
+def untar_backup_files():
+  """ Restores a previous backup into the Cassandra directory structure
+  from a tar ball. 
+
+  Raises:
+    BRException: On untar issues.
+  """
+  logging.info("Untarring Cassandra backup files...")
+  try:
+    tar = tarfile.open(BACKUP_FILE_LOCATION, "r:gz")
+    tar.extractall(path="/")
+    tar.close()
+  except tarfile.TarError, tar_error:
+    logging.exception(tar_error)
+    raise backup_exceptions.BRException(
+      "Exception while untarring Cassandra backup files.")
+  logging.info("Done untarring Cassandra backup files.")
+
+def restore_snapshots():
+  """ Restore snapshot into correct directories.
+
+  Returns:
+    True on success, False otherwise.
+  """
+  logging.info("Restoring Cassandra snapshots.")
+
+  for directory in CASSANDRA_DATA_SUBDIRS:
+    data_dir = "{0}{1}/{2}/".format(constants.APPSCALE_DATA_DIR, "cassandra",
+      directory)
+    logging.debug("Restoring in dir {0}".format(data_dir))
+    for path, _, filenames in os.walk(data_dir):
+      for filename in filenames:
+        logging.debug("Restoring: {0}".format(filename))
+        if not filename:
+          logging.warn("skipping...")
+          continue
+        full_path = "{0}/{1}".format(path, filename)
+        new_full_path = "{0}/../../{1}".format(path, filename)
+        logging.debug("{0} -> {1}".format(full_path, new_full_path))
+        # Move the files up into the data directory.
+        try:
+          call(['cp', full_path, new_full_path])
+        except CalledProcessError as error:
+          logging.error("Error while moving Cassandra snapshot in place. "
+            "Error: {0}".format(str(error)))
+          return False
+  logging.info("Done restoring Cassandra snapshots.")
+  return True
+
+def remove_local_backup_file(local_file=BACKUP_FILE_LOCATION):
+  """ Removes the local backup file.
+
+  Args:
+    local_file: A str, the path to the backup file to delete.
+  """
+  try:
+    call(['rm', '-rf', local_file])
+  except CalledProcessError as error:
+    logging.error("Error while removing local backup file '{0}'. Error: {1}".\
+      format(local_file, str(error)))
+
+def restore_data(storage, path=''):
+  """ Restores the Cassandra snapshot. 
+
+  Args:
+    storage: A str, one of the StorageTypes class members.
+    path: A str, the name of the backup file to restore from.
+  """
+  if storage == StorageTypes.GCS:
+    # Download backup file and store locally with a fixed name.
+    if not gcs_helper.download_from_bucket(path, BACKUP_FILE_LOCATION):
```

Comment:
It seems like we'd want to use a different location to avoid confusion if the system ever ends up in a broken state.